package.json

```diff
@@ -70,7 +69,8 @@
     "karma-webpack": "^2.0.3",
     "reflect-metadata": "^0.1.3",
     "rimraf": "^2.6.2",
-    "rxjs": "5.4.2",
+    "rxjs": "^6.2.0",
+    "rxjs-compat": "^6.2.0",
```

Comment:
Why do we need `rxjs-compat`? My understanding ist, that this is only needed for backwards compatibility and the aim of this pull request is to update all rxjs related code to the current version.

Reply:
But there are packages that depend on rxjs<6.0.0 and need rxjs-compat.

Reply:
I just checked it, but it seems the error message only **looks** like that there is a dependency problem.

The actual problem is, that there is still old rxjs code used:
- Usage of `ErrorObservable` instead of `Observable` (https://github.com/ghidoz/angular2-jsonapi/pull/180/files#diff-3109cc626189477fd645f91838ca75c2R323)
- Import `Observable` from `rxjs/Observable` instead of `rxjs` (in `json-api.model.ts`)

I fixed both points locally and the `servetest` script runs through. Does it work for you to?

Reply:
**Edit**: Actually I needed to update my `devDependencies` as well :/ (angular to `^6`)

I just checked it, but it seems the error message only **looks** like that there is a dependency problem.

The actual problem is, that there is still old rxjs code used:
- Usage of `ErrorObservable` instead of `Observable` (https://github.com/ghidoz/angular2-jsonapi/pull/180/files#diff-3109cc626189477fd645f91838ca75c2R323)
- Import `Observable` from `rxjs/Observable` instead of `rxjs` (in `json-api.model.ts`)

I fixed both points locally and the `servetest` script runs through. Does it work for you to?

Reply:
So, additionally I had to update the following `devDependencies`:
```
    "@angular/animations": "^6.0.4",
    "@angular/common": "^6.0.4",
    "@angular/compiler": "^6.0.4",
    "@angular/compiler-cli": "^6.0.4",
    "@angular/core": "^6.0.4",
    "@angular/platform-browser": "^6.0.4",
    "@angular/platform-browser-dynamic": "^6.0.4",
    "@angular/platform-server": "^6.0.4",
...
    "codelyzer": "~4.2.1",
....
    "typescript": "2.7.2",
....
    "zone.js": "^0.8.26"
```

@safo6m: Would it make sense to do this in this pull request (as rxjs 6.x is only usable in angular 6+)?